components/src/dynamo/planner/monitoring/dgd_services.py

```
@@ -174,12 +193,84 @@ def get_gpu_count(self) -> int:
+        """
+        annotations = (
+            self.service.get("podTemplate", {}).get("metadata", {}).get("annotations")
+            or {}
+        )
+        raw = annotations.get(POWER_ANNOTATION_KEY)
+        if raw is None:
+            raise PowerAnnotationMissingError(self.name)
+        try:
+            watts = int(str(raw).strip())
+        except (ValueError, TypeError):
+            raise PowerAnnotationInvalidError(self.name, str(raw))
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)